###Load data


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/project_1/Amazon-Product-Reviews.csv')

In [ ]:
df.head(5)

,marketplace,customer_id,review_id,product_id,product_parent,product_title,product_category,star_rating,helpful_votes,total_votes,vine,verified_purchase,review_headline,review_body,review_date,sentiment
0,US,11555559,R1QXC7AHHJBQ3O,B00IKPX4GY,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,5,0,0,N,Y,Five Stars,Great love it,31-08-2015,1
1,US,31469372,R175VSRV6ZETOP,B00IKPYKWG,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,3,0,0,N,N,Lots of ads Slow processing speed Occasionally...,Lots of ads<br />Slow processing speed<br />Oc...,31-08-2015,0
2,US,26843895,R2HRFF78MWGY19,B00IKPW0UA,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,5,0,0,N,Y,Well thought out device,Excellent unit. The versatility of this table...,31-08-2015,1
3,US,19844868,R8Q39WPKYVSTX,B00LCHSHMS,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,4,0,0,N,N,Not all apps/games we were looking forward to ...,I bought this on Amazon Prime so I ended up bu...,31-08-2015,1
4,US,1189852,R3RL4C8YP2ZCJL,B00IKPZ5V6,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,5,0,0,N,Y,Five Stars,All Amazon products continue to meet my expect...,31-08-2015,1


In [ ]:
df.shape

(30846, 16)

In [ ]:
df.describe()

,customer_id,product_parent,star_rating,helpful_votes,total_votes,sentiment
count,3.084600e+04,30846.0,30846.000000,30846.00000,30846.000000,30846.000000
mean,2.471002e+07,2693241.0,4.336154,0.68641,0.895027,0.835343
std,1.611172e+07,0.0,1.120716,30.74337,32.356354,0.370876
min,1.134600e+04,2693241.0,1.000000,0.00000,0.000000,0.000000
25%,1.150637e+07,2693241.0,4.000000,0.00000,0.000000,1.000000
50%,2.294025e+07,2693241.0,5.000000,0.00000,0.000000,1.000000
75%,4.008874e+07,2693241.0,5.000000,0.00000,0.000000,1.000000
max,5.309351e+07,2693241.0,5.000000,3720.00000,3875.000000,1.000000


In [ ]:
#Check Missing Values
df.isnull().sum()

,0
marketplace,0
customer_id,0
review_id,0
product_id,0
product_parent,0
product_title,0
product_category,0
star_rating,0
helpful_votes,0
total_votes,0


In [ ]:
#drop null value
df.dropna(subset=['review_headline', 'review_body'], inplace=True)

In [ ]:
df.isnull().sum()

,0
marketplace,0
customer_id,0
review_id,0
product_id,0
product_parent,0
product_title,0
product_category,0
star_rating,0
helpful_votes,0
total_votes,0


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30840 entries, 0 to 30845
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   marketplace        30840 non-null  object
 1   customer_id        30840 non-null  int64 
 2   review_id          30840 non-null  object
 3   product_id         30840 non-null  object
 4   product_parent     30840 non-null  int64 
 5   product_title      30840 non-null  object
 6   product_category   30840 non-null  object
 7   star_rating        30840 non-null  int64 
 8   helpful_votes      30840 non-null  int64 
 9   total_votes        30840 non-null  int64 
 10  vine               30840 non-null  object
 11  verified_purchase  30840 non-null  object
 12  review_headline    30840 non-null  object
 13  review_body        30840 non-null  object
 14  review_date        30840 non-null  object
 15  sentiment          30840 non-null  int64 
dtypes: int64(6), object(10)
memory usage: 4.0+ MB

In [ ]:
df['review_date'] = pd.to_datetime(df['review_date'], format='%d-%m-%Y')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30840 entries, 0 to 30845
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   marketplace        30840 non-null  object        
 1   customer_id        30840 non-null  int64         
 2   review_id          30840 non-null  object        
 3   product_id         30840 non-null  object        
 4   product_parent     30840 non-null  int64         
 5   product_title      30840 non-null  object        
 6   product_category   30840 non-null  object        
 7   star_rating        30840 non-null  int64         
 8   helpful_votes      30840 non-null  int64         
 9   total_votes        30840 non-null  int64         
 10  vine               30840 non-null  object        
 11  verified_purchase  30840 non-null  object        
 12  review_headline    30840 non-null  object        
 13  review_body        30840 non-null  object        
 14  review_date

In [ ]:
import re

def clean_text(text):
    if pd.isna(text):
        return ""
    text = text.lower()
    text = re.sub(r'[^a-z0-9\s]', '', text)
    return text.strip()

df['review_body_clean'] = df['review_body'].apply(clean_text)
df['review_headline_clean'] = df['review_headline'].apply(clean_text)

In [ ]:
#Memory optimization helpful for 30k rows for group by
cat_cols = ['marketplace', 'vine', 'verified_purchase', 'product_category']
df[cat_cols] = df[cat_cols].astype('category')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30840 entries, 0 to 30845
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   marketplace            30840 non-null  category      
 1   customer_id            30840 non-null  int64         
 2   review_id              30840 non-null  object        
 3   product_id             30840 non-null  object        
 4   product_parent         30840 non-null  int64         
 5   product_title          30840 non-null  object        
 6   product_category       30840 non-null  category      
 7   star_rating            30840 non-null  int64         
 8   helpful_votes          30840 non-null  int64         
 9   total_votes            30840 non-null  int64         
 10  vine                   30840 non-null  category      
 11  verified_purchase      30840 non-null  category      
 12  review_headline        30840 non-null  object        
 13  review

In [ ]:
df.duplicated().sum()

np.int64(0)

In [ ]:
df['review_id'].duplicated().sum()

np.int64(0)

In [ ]:
df['star_rating'].value_counts()

,count
star_rating,
5,20017
4,5746
3,2216
1,1707
2,1154


In [ ]:
df['sentiment'].value_counts()

,count
sentiment,
1,25763
0,5077


In [ ]:
df['verified_purchase'].value_counts()

,count
verified_purchase,
Y,26842
N,3998


In [ ]:
df['vine'].value_counts()

,count
vine,
N,30840


In [ ]:
#Total products
df['product_id'].nunique()

21

In [ ]:
#Total Categories
df['product_category'].nunique()

1

In [ ]:
df['customer_id'].nunique()

30839

In [ ]:
#average rating
df['star_rating'].mean()

np.float64(4.336316472114137)

In [ ]:
#date range
df['review_date'].min() #Timestamp('2014-10-03 00:00:00')
df['review_date'].max() #Timestamp('2015-08-31 00:00:00')

Timestamp('2015-08-31 00:00:00')

In [ ]:
#Total healpfull voter
df['helpful_votes'].sum()

np.int64(21173)

In [ ]:
#positive review
(df['sentiment']==1).mean()*100

np.float64(83.53761348897535)

In [ ]:
(df['sentiment']==0).mean()*100

np.float64(16.462386511024643)

In [ ]:
#purched rate
(df['verified_purchase']=='Y').mean()*100

np.float64(87.03631647211414)

###Dataset Summary

*   Total Customers -	30839
*   Total Sellers- 21
*   Total Categories - 1
*   Average Rating	- 4.33
*   Positive Review - 83.53%
*   Negative Review - 16.46%
*   Purchase rate - 87%
*   Date Range -	2014-10-03 to 2015-08-31
*   Missing Values - 0.01%

##Key Findings

Several important observations were identified during the data understanding phase:

* Customer satisfaction appears to be high, with an average rating of 4.33 out of 5.
* Positive reviews account for over 83% of all reviews.
* Most reviewers are verified purchasers, representing 87% of total reviews.
* The dataset contains reviews for 21 products within a single category.
* Data quality is excellent, with almost no missing values after cleaning.
* The review period spans approximately 11 months, enabling monthly trend analysis.

In [ ]:
df.head(5)

,marketplace,customer_id,review_id,product_id,product_parent,product_title,product_category,star_rating,helpful_votes,total_votes,vine,verified_purchase,review_headline,review_body,review_date,sentiment,review_body_clean,review_headline_clean
0,US,11555559,R1QXC7AHHJBQ3O,B00IKPX4GY,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,5,0,0,N,Y,Five Stars,Great love it,2015-08-31,1,great love it,five stars
1,US,31469372,R175VSRV6ZETOP,B00IKPYKWG,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,3,0,0,N,N,Lots of ads Slow processing speed Occasionally...,Lots of ads<br />Slow processing speed<br />Oc...,2015-08-31,0,lots of adsbr slow processing speedbr occasion...,lots of ads slow processing speed occasionally...
2,US,26843895,R2HRFF78MWGY19,B00IKPW0UA,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,5,0,0,N,Y,Well thought out device,Excellent unit. The versatility of this table...,2015-08-31,1,excellent unit the versatility of this tablet...,well thought out device
3,US,19844868,R8Q39WPKYVSTX,B00LCHSHMS,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,4,0,0,N,N,Not all apps/games we were looking forward to ...,I bought this on Amazon Prime so I ended up bu...,2015-08-31,1,i bought this on amazon prime so i ended up bu...,not all appsgames we were looking forward to u...
4,US,1189852,R3RL4C8YP2ZCJL,B00IKPZ5V6,2693241,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",PC,5,0,0,N,Y,Five Stars,All Amazon products continue to meet my expect...,2015-08-31,1,all amazon products continue to meet my expect...,five stars


In [ ]:
columns_to_keep = [
    'customer_id',
    'product_id',
    'product_title',
    'star_rating',
    'helpful_votes',
    'total_votes',
    'verified_purchase',
    'review_date',
    'sentiment',
    'review_body_clean',
    'review_headline_clean'
]

df_eda = df[columns_to_keep].copy()

In [ ]:
df_eda['year'] = df_eda['review_date'].dt.year
df_eda['month'] = df_eda['review_date'].dt.month
df_eda['month_name'] = df_eda['review_date'].dt.strftime('%b')

In [ ]:
df_eda.head(1)

,customer_id,product_id,product_title,star_rating,helpful_votes,total_votes,verified_purchase,review_date,sentiment,review_body_clean,review_headline_clean,year,month,month_name
0,11555559,B00IKPX4GY,"Fire HD 7, 7"" HD Display, Wi-Fi, 8 GB",5,0,0,Y,2015-08-31,1,great love it,five stars,2015,8,Aug


In [ ]:
df_eda.to_excel(
    'customer_reviews_cleaned.xlsx',
    index=False
)

In [ ]:
from google.colab import files

files.download('customer_reviews_cleaned.xlsx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>